scipy/stats/stats.py

```diff
@@ -4153,6 +4154,179 @@ def binary_search(n, n1, n2, side):
     return oddsratio, pvalue
 
 
+def _barnard_exact_compute_gammaln_combination(n):
+    """Compute all log combination of C(n, k)"""
+    gammaln_arr = special.gammaln(np.arange(n + 1) + 1)
+    return special.gammaln(n + 1) - gammaln_arr - gammaln_arr[::-1]
+
+
+def barnard_exact(
+        table,
+        alternative='two-sided',
+        pooled=True,
+        num_it=4):
+    """
+    Perform a Barnard exact test on a 2x2 contingency table.
+
+    Parameters
+    ----------
+    table : array_like of ints
+        A 2x2 contingency table.  Elements should be non-negative integers.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+          * 'two-sided'
+          * 'less': one-sided
+          * 'greater': one-sided
+
+    pooled : bool, optional
+        Whether to compute score statistic with pooled variance or
+        unpooled (Wald statistic)
+
+    num_it : int, optional
+        Number of iterations of the grid search. Default is 4. Must be
+        non-negative.
+
+    Returns
+    -------
+    score_statistic : float
+        Z statistic with pooled (Score) or unpooled (Wald) variance, depending
+        on the user choice of `pooled` param.
+    p_value : float
+        P-value, the probability of obtaining a distribution at least as
```

Comment:
Consider "distribution" -> "test statistic"

Reply:
Resolved in df7fa9d13b5ad212f53198d903218cb71a189ff4